plugins/cue_points/base/lib/kCuePointManager.php

```diff
@@ -1,11 +1,80 @@
 <?php
+
+class KAMFData
+{
+	public $pts;
+	public $ts;
+
+};
+
 /**
  * @package plugins.cuePoint
  */
-class kCuePointManager implements kObjectDeletedEventConsumer, kObjectChangedEventConsumer, kObjectAddedEventConsumer, kObjectCreatedEventConsumer, kObjectReplacedEventConsumer
+class kCuePointManager implements kBatchJobStatusEventConsumer, kObjectDeletedEventConsumer, kObjectChangedEventConsumer, kObjectAddedEventConsumer, kObjectReplacedEventConsumer
 {
 	const MAX_CUE_POINTS_TO_COPY_TO_VOD = 100;
 	const MAX_CUE_POINTS_TO_COPY_TO_CLIP = 1000;
+	const CUE_POINT_TIME_EPSILON = 1;
+
+	/* (non-PHPdoc)
+ 	 * @see kBatchJobStatusEventConsumer::updatedJob()
+ 	 */
+	public function updatedJob(BatchJob $dbBatchJob)
+	{
+		KalturaLog::debug('in kCuePointManager.updatedJob - jobID:' . $dbBatchJob->getId() . ' job type: ' . $dbBatchJob->getJobType()  . ' JobStatus= ' . $dbBatchJob->getStatus());
+
+		if ($jobType = $dbBatchJob->getJobType() == BatchJobType::CONCAT){
+			self::handleConcatJobFinished($dbBatchJob, $dbBatchJob->getData());
+		}
+		else if ($jobType = $dbBatchJob->getJobType() == BatchJobType::CONVERT_LIVE_SEGMENT) {
+			self::handleConvertLiveSegmentJobFinished($dbBatchJob, $dbBatchJob->getData());
+		}
+		return true;
+	}
+
+	private function handleConvertLiveSegmentJobFinished(BatchJob $dbBatchJob, kConvertLiveSegmentJobData $data)
+	{
+		KalturaLog::debug('in kCuePointManager.handleConvertLiveSegmentJobFinished - file index is ' . $data->getFileIndex());
+		if ($data->getFileIndex() == 0) {
+			$mediaInfoParser = new KMediaInfoMediaParser($data->getDestFilePath(), kConf::get('bin_path_mediainfo'));
```

Comment:
Why the parsing is needed here if it's already done before the job (or during the job as I suggested)?